app/models/prisoner.rb

```diff
@@ -6,13 +6,19 @@ class Prisoner
   attribute :prison_name, String
 
   validates_format_of :number, with: /\A[a-z]\d{4}[a-z]{2}\z/i, message: "must be a valid prisoner number" # eg a1234aa
-  validates_inclusion_of :prison_name, in: Rails.configuration.prison_data.keys, message: "must be chosen"
+  validates_inclusion_of :prison_name, in: Prison.names, message: "must be chosen"
   validate :prison_in_service
 
   def prison_in_service
-    if !self.prison_name.blank? && !Rails.configuration.prison_data[self.prison_name]['enabled']
+    unless prison && prison.try(:enabled)
```

Comment:
`&&` and `try` is duplication, and it would be nice to use the `enabled?` method you defined above.

```ruby
unless prison.try(:enabled?)
```

or

```ruby
unless prison && prison.enabled?
```

(my preference).